### ISSUE TITLE
[Ch 03][Notebook][Minor] 삽질 코너 힌트 "LogReg 와 비슷할 수도" — 실측 약 0.79 대 0.86 (약 7%p 차)

### ISSUE THREAD

**Issue by modesta-jihye-lee** [yoon-gu/neuqes-101#96]
## 대상

- **챕터**: Ch 03 — sklearn Binary (출력에 sigmoid가 붙다)
- **산출물**: Notebook (`03_sklearn_binary/03_sklearn_binary.ipynb` — `## 🚀 삽질 코너` `cell022`)
- **기준 커밋**: `eed3dbe`
- **환경**: 정적 검토 + 로컬 실측(sklearn 1.9.0, 노트북과 동일 `seed=42`/`random_state=42`/`stratify`). **Colab 전체 실행 완료** (2026-08-15 · CPU 런타임 · 43초 · 에러 0건)

## 실제 결과

삽질 코너의 힌트가 실제 결과와 어긋납니다.

> 힌트: LPM 은 출력이 [0, 1]을 안 지키지만, 임계값 0.5로 자르면 **분류 성능 자체는 LogReg 와 비슷할 수도 있습니다.**

같은 데이터·같은 분할로 삽질 코너 코드를 그대로 돌린 결과:

| 모델 | 로컬 (sklearn 1.9.0) | Colab 재실행 (2026-08-15) |
|---|---|---|
| LPM (`LinearRegression` + 임계값 0.5) | **0.7921** | **0.7958** |
| `LogisticRegression` | **0.8639** | **0.8639** |
| 차이 | **7.2%p** | **6.8%p** |

**약 7%p 차이**로 "비슷" 하다고 보기 어렵습니다. 삽질 코너는 학습자가 **직접 돌려 보는 자리**라, 힌트를 읽고 실행하면 예상과 다른 결과를 보게 됩니다.

두 실행에서 `LogisticRegression` 의 정확도는 **0.8639 로 완전히 동일**했고 LPM 만 0.7921 → 0.7958 로 움직였습니다(Ch 1·2 와 같은 어휘 컷 차이). 어느 실행에서도 격차가 6.8%p 이상이라, **이 지적은 환경에 좌우되지 않습니다.**

원인도 이 커리큘럼 안에서 설명됩니다 — Ch 2 에서 확인한 것처럼 이 셋업은 feature 10,000 > 샘플 3,232 라, LPM 이 학습 데이터를 보간해 버려 일반화가 무너집니다.

## 기대 결과

> "힌트: LPM 은 출력이 [0, 1]을 안 지킬 뿐 아니라, 이 셋업(feature 10,000 > 샘플 3,232)에서는 정확도도 **눈에 띄게 떨어집니다** — 실행해 보면 LogReg 보다 **약 7%p 낮게** 나옵니다. 무엇보다 '출력이 확률' 이라는 해석을 잃습니다."

## 재현 절차

```python
# 노트북 삽질 코너 코드 그대로
from sklearn.linear_model import LinearRegression
from sklearn.metrics import accuracy_score
lpm = LinearRegression().fit(X_train, y_train)
pred_lpm = lpm.predict(X_test)
print(accuracy_score(y_test, (pred_lpm >= 0.5).astype(int)))   # 0.7921
print(accuracy_score(y_test, y_pred))                          # 0.8639
```

## 로그

```
# 로컬 (sklearn 1.9.0)
LPM 예측 범위: [-0.806, 1.847] / 0미만 129개 / 1초과 147개
LPM accuracy(0.5): 0.7921
LogReg accuracy  : 0.8639        (차이 7.2%p)

# Colab 재실행 (2026-08-15, CPU 런타임)
LPM 예측 범위: [-0.800, 1.838] / 0미만 126개 / 1초과 146개
LPM accuracy(0.5): 0.7958
LogReg accuracy  : 0.8639        (차이 6.8%p — LogReg 는 두 실행 완전 동일)
```

## 전파 범위

| 산출물 | 상태 |
|---|---|
| `03_sklearn_binary/03_sklearn_binary.ipynb` `cell022` | ❌ |
| `pages/03-sklearn_binary-wrapup.md` | ❌ |
| `book/chapters/ch03_sklearn_binary.tex` | ❌ → EPUB 영향 |

## 심각도

**Minor** — 선택 코너의 힌트라 본문 학습 흐름을 막지는 않습니다. 다만 학습자가 직접 돌려 보는 자리이므로 실측값을 적어 두는 편이 낫습니다. 내용 동결(9/20) 전 수정 가능합니다.

## 재실행 여부

마크다운 셀 수정이므로 **노트북 재실행은 불필요**합니다. WikiDocs 재변환 → EPUB 재빌드가 필요합니다.

## 담당

- 수정 담당자: 미정
- 재검수자: 등록자 (@modesta-jihye-lee)
- 연결: 챕터 검수 이슈 #27

**Comment by yoon-gu** [comment]
PASS — 삽질 코너 힌트가 "LogReg 와 비슷할 수도" 에서 "이 셋업(feature 10,000 > 샘플 3,232)에서는 정확도도 눈에 띄게 떨어집니다 — LogReg 보다 약 7%p 낮게 나옵니다" 로 실측화됐습니다.

— PR #289 (master `6986584`, 2026-09-18). `origin/master` 기준으로 노트북·`executed/`·`pages/`·`tex` 전파까지 확인했습니다.
